File: src/main/java/ru/arc/xserver/XPay.java
package ru.arc.xserver;

import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;
import net.milkbowl.vault.economy.EconomyResponse;
import org.bukkit.Bukkit;
import org.bukkit.OfflinePlayer;
import ru.arc.ARC;
import ru.arc.configs.Config;
import ru.arc.configs.ConfigManager;

import static ru.arc.util.Logging.debug;
import static ru.arc.util.Logging.error;

@EqualsAndHashCode(callSuper = true)
@Data
@ToString(callSuper = true)
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class XPay extends XAction {

    double amount;
    String playerName;
    UUID playerUuid;

    private static final Config miscConfig = ConfigManager.of(ARC.plugin.getDataFolder().toPath(), "misc.yml");

    @Override
    protected void runInternal() {
        if (amount == 0) {
            error("Amount is 0: {}", this);
            return;
        }
        if (!miscConfig.bool("xaction.main-server", false)) {
            debug("Main server is disabled, skipping pay action");
            return;
        }
        OfflinePlayer player = null;
        if (playerName != null) {
            player = Bukkit.getOfflinePlayer(playerName);
        } else if (playerUuid != null) {
            player = Bukkit.getOfflinePlayer(playerUuid);
        }
        if (player == null) {
            error("Player not found: {}", this);
            return;
        }
        EconomyResponse economyResponse = ARC.getEcon().depositPlayer(player, amount);
        if (!economyResponse.transactionSuccess()) {
            error("Error depositing money: {}", economyResponse.errorMessage);
        }
    }
}

File: src/main/kotlin/ru/arc/xserver/XPay.kt
package ru.arc.xserver

import org.bukkit.Bukkit
import ru.arc.ARC
import ru.arc.config.ConfigManager
import ru.arc.core.modules.EconomyModule
import ru.arc.util.Logging.debug
import ru.arc.util.Logging.error
import java.util.UUID

class XPay(
    var amount: Double = 0.0,
    var playerName: String? = null,
    var playerUuid: UUID? = null
) : XAction() {

    override fun runInternal() {
        val miscConfig = ConfigManager.of(ARC.instance.dataFolder.toPath(), "misc.yml")
        if (amount == 0.0) {
            error("Amount is 0: {}", this)
            return
        }
        if (!miscConfig.bool("xaction.main-server", false)) {
            debug("Main server is disabled, skipping pay action")
            return
        }
        val player = when {
            playerName != null -> Bukkit.getOfflinePlayer(playerName!!)
            playerUuid != null -> Bukkit.getOfflinePlayer(playerUuid!!)
            else -> null
        }
        if (player == null) {
            error("Player not found: {}", this)
            return
        }
        val economy = EconomyModule.getEconomy() ?: return
        val response = economy.depositPlayer(player, amount)
        if (!response.transactionSuccess()) {
            error("Error depositing money: {}", response.errorMessage)
        }
    }
}
